cli/npm/managed/cache.rs

```diff
@@ -127,7 +123,26 @@ impl NpmCache {
       .await?;
     match maybe_bytes {
       Some(bytes) => {
-        verify_and_extract_tarball(package, &bytes, dist, &package_folder)
+        // the user ran with `--reload`, so remove the existing directory
+        if !should_use_cache && self.fs.exists_sync(&package_folder) {
+          self
+            .fs
+            .remove_sync(&package_folder, true)
+            .map_err(AnyError::from)
+            .context("Failed removing package directory.")?;
```

Comment:
Actually, this change here is bad because it could leave a folder in a bad state if a user runs with `--reload` and kills the process during this call. I'll revert this to use the old behaviour when running with `--reload`.